File: app/src/main/java/com/example/rahmatantravel/belumLunasPesanan.java
package com.example.rahmatantravel;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.rahmatantravel.Adapter.PesananAdapter;
import com.example.rahmatantravel.Models.PesananModels;

import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Date;

/**
 * A simple {@link Fragment} subclass.
 * Use the {@link belumLunasPesanan#newInstance} factory method to
 * create an instance of this fragment.
 */
public class belumLunasPesanan extends Fragment {

    // TODO: Rename parameter arguments, choose names that match
    // the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
    private static final String ARG_PARAM1 = "param1";
    private static final String ARG_PARAM2 = "param2";

    // TODO: Rename and change types of parameters
    private String mParam1;
    private String mParam2;
    private RecyclerView recycleBelumLunasPesanan;
    private PesananAdapter pesananAdapter2;
    private ArrayList<PesananModels> pesananModelsArrayList2;

    public belumLunasPesanan() {
        // Required empty public constructor
    }

    /**
     * Use this factory method to create a new instance of
     * this fragment using the provided parameters.
     *
     * @param param1 Parameter 1.
     * @param param2 Parameter 2.
     * @return A new instance of fragment belumLunasPesanan.
     */
    // TODO: Rename and change types and number of parameters
    public static belumLunasPesanan newInstance(String param1, String param2) {
        belumLunasPesanan fragment = new belumLunasPesanan();
        Bundle args = new Bundle();
        args.putString(ARG_PARAM1, param1);
        args.putString(ARG_PARAM2, param2);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mParam1 = getArguments().getString(ARG_PARAM1);
            mParam2 = getArguments().getString(ARG_PARAM2);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_belum_lunas_pesanan, container, false);
        addDataBelumLunasPesanan();
        recycleBelumLunasPesanan = (RecyclerView) view.findViewById(R.id.recycleBelumLunasPesanan);
        pesananAdapter2 = new PesananAdapter(pesananModelsArrayList2);
        RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(getContext(), LinearLayoutManager.VERTICAL,false);

        if (pesananAdapter2 == null){
            pesananAdapter2 = new PesananAdapter(pesananModelsArrayList2);
        }
        recycleBelumLunasPesanan.setLayoutManager(layoutManager);
        recycleBelumLunasPesanan.setAdapter(pesananAdapter2);

        return view;
    }
    void addDataBelumLunasPesanan(){
        pesananModelsArrayList2 = new ArrayList<>();
        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");

        try {
            Date date1 = dateFormat.parse("2023-01-21");
            Date date2 = dateFormat.parse("2023-06-10");
            Date date3 = dateFormat.parse("2023-01-04");
            Date date4 = dateFormat.parse("2023-04-08");
            Date date5 = dateFormat.parse("2023-02-10");


            pesananModelsArrayList2.add(new PesananModels("RMTN1423132", "Paket Umroh 21 Nov 2023", date1, "BSI", "Belum Lunas"));
            pesananModelsArrayList2.add(new PesananModels("RMTN1423133", "Paket Umroh 13 Okt 2023", date2, "BSI", "Belum Lunas"));
            pesananModelsArrayList2.add(new PesananModels("RMTN1423134", "Paket Umroh 05 Des 2023", date3, "BSI", "Belum Lunas"));
            pesananModelsArrayList2.add(new PesananModels("RMTN1423135", "Paket Umroh 01 Jun 2023", date4, "BSI", "Belum Lunas"));
            pesananModelsArrayList2.add(new PesananModels("RMTN1423136", "Paket Umroh 14 Jul 2023", date5, "BSI", "Belum Lunas"));

            PesananAdapter adapter2 = new PesananAdapter(pesananModelsArrayList2);
            recycleBelumLunasPesanan.setAdapter(adapter2);
        }catch (Exception e){
            e.printStackTrace();
            Log.e("DateError", "Error parsing date: " + e.getMessage());
        }
    }
}

File: app/src/main/java/com/example/rahmatantravel/belumLunasPesanan.kt
package com.example.rahmatantravel;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView

import com.example.rahmatantravel.Adapter.PesananAdapter;
import com.example.rahmatantravel.Models.PesananModels;
import com.example.rahmatantravel.SharedPref.PrefManager
import com.example.rahmatantravel.api.RetrofitClient
import com.example.rahmatantravel.api.pemesananResponse.APIPesanan
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

import java.util.ArrayList;

class belumLunasPesanan : Fragment() {

    private lateinit var recyclerView: RecyclerView
    private lateinit var imageView: ImageView

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_belum_lunas_pesanan, container, false)

        recyclerView = view.findViewById(R.id.recyclerView)
        imageView = view.findViewById(R.id.imageView)

        // Sembunyikan imageView di awal
        imageView.visibility = View.GONE

        val prefManager = PrefManager(requireContext())
        val loginData = prefManager.getLoginData()

        val userID = loginData.first ?.toInt()
        logData("User ID", userID.toString())

        val layoutManager = LinearLayoutManager(context, LinearLayoutManager.VERTICAL, false)

        RetrofitClient.instance.getPemesananByStatus("belumLunas", "$userID").enqueue(
            object  : Callback<APIPesanan> {
                override fun onResponse(
                    call: Call<APIPesanan>,
                    response: Response<APIPesanan>
                ) {
                    if (response.isSuccessful){
                        if (response.body() != null){
                            val responseBody = response.body()
                            logData("Response Success", responseBody ?.data.toString())
                            val data = responseBody ?.data
                            if (data.isNullOrEmpty()) {
                                // Jika data kosong, tampilkan imageView dan sembunyikan recyclerView
                                imageView.visibility = View.VISIBLE
                                recyclerView.visibility = View.GONE
                            } else {
                                // Jika data tidak kosong, tampilkan recyclerView dan sembunyikan imageView
                                imageView.visibility = View.GONE
                                recyclerView.visibility = View.VISIBLE
                                // Update recyclerView dengan data
                                recyclerView = view.findViewById(R.id.recyclerView)
                                imageView = view.findViewById(R.id.imageView)

                                // Sembunyikan imageView di awal
                                imageView.visibility = View.GONE
                                val pesananAdapter = data?.let { PesananAdapter(it) }
                                recyclerView.layoutManager = layoutManager
                                recyclerView.adapter = pesananAdapter
                            }
                        }else{
                            logData("Response Body Null", "Body is null")
                        }
                    }else{
                        val responseBody = response.body()
                        logData("Response Failed", response.message())
                        logData("Response Failed", responseBody ?.status.toString())
                    }
                }

                override fun onFailure(call: Call<APIPesanan>, t: Throwable) {
                    logData("Response on Failure", t.message.toString())
                }

            }
        )

        return view
    }

    private fun logData(tag: String, data: String) {
        Log.d(tag, "$tag: $data")
    }
}
